File: Jongryul/04/11478.py
import sys
s = sys.stdin.readline().rstrip()

st = set()

for i in range(len(s)):
    tmp = s[i]
    st.add(tmp)
    for j in range(i+1, len(s)):
        tmp += s[j]
        st.add(tmp)
print(len(st))

File: Jongryul/04/11478.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const s = fs.readFileSync(filePath).toString().trim();

const set = new Set(s.split(""));

for (let i = 0; i < s.length; i++) {
  let tmp = s[i];
  for (let j = i + 1; j < s.length; j++) {
    tmp += s[j];
    if (!set.has(tmp)) {
      set.add(tmp);
    }
  }
}

console.log(set.size);
